frontend/src/scenes/dashboard/DashboardTemplateVariables.tsx

```diff
@@ -0,0 +1,61 @@
+import { LemonLabel } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { useEffect } from 'react'
+import { ActionFilter } from 'scenes/insights/filters/ActionFilter/ActionFilter'
+import { FilterType, InsightType } from '~/types'
+import { dashboardTemplateVariablesLogic } from './DashboardTemplateVariablesLogic'
+import { newDashboardLogic } from './newDashboardLogic'
+
+import './DashboardTemplateVariables.scss'
+
+export function DashboardTemplateVariables(): JSX.Element {
+    const { activeDashboardTemplate } = useValues(newDashboardLogic)
+
+    const { variables } = useValues(dashboardTemplateVariablesLogic)
+    const { setVariables, setVariable } = useActions(dashboardTemplateVariablesLogic)
+
+    const FALLBACK_EVENT = {
+        id: '$pageview',
+        math: 'dau',
+        type: 'events',
+    }
+
+    useEffect(() => {
+        setVariables(activeDashboardTemplate?.variables || [])
+    }, [activeDashboardTemplate])
```

Comment:
Instead of this `useEffect` dance, you can also pass `activeDashboardTemplate` as a prop to the logic, and use `propsChanged` within the logic to trigger this action. This makes it a bit cleaner, but also faster, as this useEffect triggers only after the first time the component renders.